client/tailscale/localclient.go

```diff
@@ -391,16 +391,26 @@ func (lc *LocalClient) DebugAction(ctx context.Context, action string) error {
 	return nil
 }
 
+// DebugPortmapOpts contains options for the DebugPortmap command.
+type DebugPortmapOpts struct {
+	// LogHTTP instructs the debug-portmap endpoint to print all HTTP
+	// requests and responses made to the logs.
+	LogHTTP bool
+}
+
 // DebugPortmap invokes the debug-portmap endpoint, and returns an
```

Comment:
And the IP should be a netip.Addr probably and required.

Reply:
> what is duration used for? I guess how long to make the mapping?

Yep!

> Maybe we should just change this signature entirely to only take big opts, or make everything plausibly optional into opts, like duration (default to 1 minute or something).

I could do that, yeah; I like it better than "a bucket of strings".

> What is "ty"? Can it have a default?

From reading the code... it's whether we want to perform a specific kind of portmap test; empty means "all", otherwise you can specify "pmp", "pcp" or "upnp". Will document.

> And the IP should be a netip.Addr probably and required.

`gwSelf` isn't required because it will be determined via `netMon.GatewayAndSelfIP` if not provided; will document.

Reply:
@bradfitz - rewrote this method to take a (documented) struct type. I intentionally didn't change the format of the requests to `tailscaled` so as not to completely break backwards compatibility, but at least all the fields in the struct are documented and properly typed (i.e. not `gatewayAndSelf string // format "gatewayAddr/selfAddr"`)